### ISSUE TITLE
Failed to get data

### ISSUE THREAD

**Issue by davidmarothi** [brunorosilva/todoist-analytics#4]
I try to use the herokuapp hosted app, but after a few sec of waiting I get the following error:

> KeyError: "['inbox_project'] not in index"
Traceback:
File "/usr/local/lib/python3.7/site-packages/streamlit/script_runner.py", line 354, in _run_script
    exec(code, module.__dict__)
File "/app/streamlit_app.py", line 3, in <module>
    app = create_app()
File "/app/todoist_analytics/frontend/dashboard.py", line 34, in create_app
    completed_tasks, active_tasks = get_data(token)
File "/usr/local/lib/python3.7/site-packages/streamlit/legacy_caching/caching.py", line 576, in wrapped_func
    return get_or_create_cached_value()
File "/usr/local/lib/python3.7/site-packages/streamlit/legacy_caching/caching.py", line 558, in get_or_create_cached_value
    return_value = func(*args, **kwargs)
File "/app/todoist_analytics/backend/utils.py", line 20, in get_data
    dc._collect_all_completed_tasks()
File "/app/todoist_analytics/backend/data_collector.py", line 46, in _collect_all_completed_tasks
    self._collect_completed_tasks(limit=200, offset=self.current_offset)
File "/app/todoist_analytics/backend/data_collector.py", line 26, in _collect_completed_tasks
    self._append_to_properties(data)
File "/app/todoist_analytics/backend/data_collector.py", line 31, in _append_to_properties
    pd.DataFrame.from_dict(data["projects"], orient="index"),
File "/app/todoist_analytics/backend/data_collector.py", line 97, in _preprocess_completed_tasks
    projects[["project_id", "name", "color", "inbox_project"]],
File "/usr/local/lib/python3.7/site-packages/pandas/core/frame.py", line 3464, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1)[1]
File "/usr/local/lib/python3.7/site-packages/pandas/core/indexing.py", line 1314, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis)
File "/usr/local/lib/python3.7/site-packages/pandas/core/indexing.py", line 1377, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")

any clue? :) 
PS: I'm a heavy Todoist user with a lot of historical data

**Comment by brunorosilva** [comment]
Hey, David

Thanks for notifying, I'll check on it and follow up when it's solved.

**Comment by brunorosilva** [comment]
@davidmarothi could you check if the issue has been fixed?

**Comment by davidmarothi** [comment]
I tried, and it worked! It took some tie to load in. I suspect there is a 10k limit for API usage? I got statistics "only" for 10k completed tasks (I have 60k+ lifetime tasks 😄 ).
I'll definitely keep an eye on the progress of this project going forward, and report any issue I see.

Thanks for handling the issue in such a short time 👍

**Comment by brunorosilva** [comment]
Great!

The API actually has a request limit of 200 tasks per request and 7 to 8 requests per 5s (~1.6k tasks/5s), for this MVP I fixed the amount of data to 10k tasks. 

I began to develop a "get more data" button, but Streamlit is not collaborating with me. In the next version (flask + react) I'll implement this in a more scalable way.

Thanks for using and any feedback or feature request is welcome.
